internal/events/webhooks/webhooks.go

```diff
@@ -444,6 +444,168 @@ func (wh *WebHooks) doDelivery(ctx context.Context, connID string, reply bool, s
 	}
 }
 
+func (wh *WebHooks) attemptBatchRequest(ctx context.Context, sub *core.Subscription, events []*core.EventDelivery, data []core.DataArray) (req *whRequest, res *whResponse, err error) {
```

Comment:
Does this not need to be an array of event+data elements? (rather than separate arrays of events, and data)

I believe in the `batch=true` case, the payload of each JSON object inside the array should be identical to the payload of the whole `POST` payload in the `batch=false` ase

Reply:
I guess key point from me, is I expect to see exactly one function that implements this logic, and the surrounding function to be broken apart to either package a single one of these into a HTTP request, or an array of them:

https://github.com/hyperledger/firefly/blob/816463cc5686d0fdc96241a9b6a41fe57e626003/internal/events/webhooks/webhooks.go#L324-L340

Reply:
I see , I can merge both which makes sense but I can't modify the request components that are shared such as the headers, path, query parameters, etc... 

So the input mapping with the firstData which is done today will only be able to be done once

Reply:
Think tis is all resolved now 👍